### DISCUSSION TITLE
migrate / db push command not generate id as postgres serial column

### DISCUSSION THREAD

**Discussion by mustofa-id** [prisma/prisma#8292] [Q&A] [answered]
My Prisma schema:
![image](https://user-images.githubusercontent.com/25121822/125728157-2cd41cb5-a96f-4a02-9339-df52818b13b6.png)

DDL result:
![image](https://user-images.githubusercontent.com/25121822/125728277-e7de453d-aacb-4dce-8604-4f89c3f74de0.png)

But, when I manually create table with `serial` column from dbms then pull it using Prisma command, it's generate same as my person model.

I missing something?

    Prisma version: ^2.27.0
    Postgres version: 13.3
    Node version: v14.17.0

**Comment by tomhoule** [comment]
The DDL migrate generates would be `CREATE TABLE ( id SERIAL PRIMARY KEY );` indeed. Identity columns are only available in postgres 9.5+ (if my memory serves right), but prisma supports all versions back to 9.4. There is an issue in prisma/prisma about using SERIAL for later versions, but this is risky, since many people develop on a different version from what they deploy to.

**Reply by ryands17** [reply]
@mustofa-id 👋 

I also cannot reproduce with the same versions of Prisma and Postgres. I used a simple schema:

```prisma
model User {
  id   Int    @id @default(autoincrement())
  name String
}
```

and here's the generated migration:
![image](https://user-images.githubusercontent.com/19697099/125736723-cde15d68-7001-48f8-93ac-db5216e91492.png)

**Comment by mustofa-id** [comment, marked as answer]
OK, after reading Postgres [docs](https://www.postgresql.org/docs/8.1/datatype.html#DATATYPE-SERIAL), it says SERIAL is not a true data type, but is simply shorthand notation that tells Postgres to create an ID column.

My screenshot above is from `pgadmin` and here from `dbeaver`:
![Screenshot_20210722_151616](https://user-images.githubusercontent.com/25121822/126610164-c73ec9fd-8218-4730-a925-8d57d1b91dfc.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
